### PR TITLE
First Rust release

### PR DESCRIPTION
**Pull request by Nima-Rahimi-Foroushaani**
Integration of the `Rust` support in `VeriFast` with the main `VeriFast` repository.

### CODE REVIEW

**Comment by btj** [comment]
The squash commit message should mention that this commit also adds support for `auto x = E;`.

File: src/win/SetErrorMode.ml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+external set_error_mode: unit -> unit = "caml_SetErrorMode"
+
+let () =
```
Comment (btj) [inline src/win/SetErrorMode.ml:3]: So this is simply being ignored, right? This does not end up in the executable?

File: testsuite.mysh [hunk 2]
```diff
@@ -109,7 +110,8 @@ cd examples
     verifast_both -c gotsmanlock.c
     verifast_both -c spinlock.c
     verifast -c -target 32bit concurrentqueue.c
-    verifast -c -prover z3v4.5 -target 32bit -fno-strict-aliasing -assume_no_provenance concurrentstack.c
+    # The following test failed after including `list.gh` in the `listex.gh`
+    # verifast -c -prover z3v4.5 -target 32bit -fno-strict-aliasing -assume_no_provenance concurrentstack.c
```
Comment (btj) [inline testsuite.mysh:114]: It'd be nice if we could rescue it. I will look at it.

**Review by btj — CHANGES_REQUESTED** [review]
Looks good!! Just a few changes necessary.

File: testsuite.mysh (hunk 2 above)
Comment (btj) [reply testsuite.mysh:114]: Fixed: https://github.com/Nima-Rahimi-Foroushaani/verifast/pull/5

File: src/win/SetErrorMode.ml (hunk 1 above)
Comment (Nima-Rahimi-Foroushaani) [reply src/win/SetErrorMode.ml:3]: Yes, it is not getting built for now.

File: bin/listex.gh [hunk 3]
```diff
@@ -237,4 +239,11 @@ lemma void foreachp_join<t>(list<t> xs, list<t> ys);
     requires foreachp<t>(xs, ?p) &*& foreachp<t>(ys, p);
     ensures foreachp<t>(append(xs, ys), p);
 
+fixpoint option<t> get_pred<t>(fixpoint(t, bool) f, list<t> xs) {
```
[the remark this answers is not in the document]
Comment (Nima-Rahimi-Foroushaani) [reply bin/listex.gh:242]: It is used by iris masks, so is not necessary in this release.
I am going to remove it.

File: .gitignore [hunk 4]
```diff
@@ -39,6 +39,11 @@
 /src/a.out
 /src/pa_verifast.cmo
 /src/vfconfig.ml
+/bin/vf-rust-mir-exporter
+
+# release files
+/src/verifast-*
```
[the remark this answers is not in the document]
Comment (Nima-Rahimi-Foroushaani) [reply .gitignore:44]: The following lines were added back then to ignore built artefacts
```
/src/verifast-*
/upload
```

**Review by btj — APPROVED** [review]

**Pull request merged**